automata/fa/nfa.py

```diff
@@ -216,21 +217,165 @@ def from_regex(
         Self
             The NFA accepting the language of the input regex.
         """
+        # Import the shorthand character classes
+        from automata.regex.parser import (
+            DIGIT_CHARS,
+            NON_DIGIT_CHARS,
+            NON_WHITESPACE_CHARS,
+            NON_WORD_CHARS,
+            WHITESPACE_CHARS,
+            WORD_CHARS,
+        )
```

Comment:
@StefanosChaliasos Can you please keep all imports at the top of the file? There's no particular need for the tighter scoping here, IMO.

cc @eliotwrobson

Reply:
Yes, I think ruff will complain about the imports.